src/Components/Components/src/Routing/Router.cs

```diff
@@ -382,23 +388,43 @@ private void OnNotFound(object sender, EventArgs args)
         if (_renderHandle.IsInitialized)
         {
             Log.DisplayingNotFound(_logger);
-            _renderHandle.Render(builder =>
+            RenderNotFound();
+        }
+    }
+
+    private void RenderNotFound()
+    {
+        _renderHandle.Render(builder =>
+        {
+            if (NotFoundPage != null)
             {
-                if (NotFoundPage != null)
+                if (_notFoundLayoutType is Type layoutType)
                 {
-                    builder.OpenComponent(0, NotFoundPage);
+                    // Directly instantiate the layout type, supplying the NotFoundPage as the Body
+                    builder.OpenComponent(0, layoutType);
+                    builder.AddAttribute(1, LayoutComponentBase.BodyPropertyName,
+                        (RenderFragment)(childBuilder =>
+                        {
+                            childBuilder.OpenComponent(2, NotFoundPage);
+                            childBuilder.CloseComponent();
+                        }));
                     builder.CloseComponent();
                 }
-                else if (NotFound != null)
-                {
-                    NotFound(builder);
-                }
                 else
                 {
-                    DefaultNotFoundContent(builder);
+                    builder.OpenComponent(0, NotFoundPage);
+                    builder.CloseComponent();
```

Comment:
Instead of manually searching for the layout and creating this whole structure, why don't we render a `RouteView` component and pass in the `_notfoundPageType` (and possibly empty parameters).

That allow us to reuse the logic and makes sure that if we ever need to change it, is in one place.